### PR TITLE
LB-386: MessyBrainz: Check corner cases for empty MBID fields.

### PR DESCRIPTION
**Pull request by kartikeyaSh**
# Summary
Add conditions to ignore recordings with empty MBIDs
We have recordings in recording_json with MBIDs pointing to
empty strings or empty lists which should be ignored.

* This is a…
    * [x] Bug fix
    * [ ] Feature addition
    * [ ] Refactoring
    * [x] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem
In recording JSON table we have recording_mbid='' and artist_mbids='[]' and release_mbid = ''. And also release which don't contain release name but contain release MBID. So, we need to handle these cases.

* JIRA ticket (_optional_): [LB-386](https://tickets.metabrainz.org/browse/LB-386)

# Solution

Modify queries to handle those cases.

### CODE REVIEW

**Review by paramsingh — COMMENTED** [review]
i think this can be merged, after our discussion on testing stuff first

File: messybrainz/db/recording.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def fetch_distinct_recording_mbids(connection):
         SELECT DISTINCT rj.data ->> 'recording_mbid'
                    FROM recording_json AS rj
               LEFT JOIN recording_cluster AS rc
-                     ON (rj.data ->> 'recording_mbid')::uuid = rc.recording_gid
+                     ON (rj.data ->> 'recording_mbid') = rc.recording_gid::text
```
Comment (mayhem) [inline messybrainz/db/recording.py:68]: Why this change? If PG can compare UUIDs, I would expect that to be faster than converting the UUID to text and then comparing it.

**Review by mayhem — COMMENTED** [review]
Overall looks sane, but I have one question...

File: messybrainz/db/recording.py (hunk 1 above)
Comment (kartikeyaSh) [reply messybrainz/db/recording.py:68]: thats for the speed. if i do it other way i.e. data ->>'recording_mbids'::uuid === raj.recording_mbid it gets slow

**Comment by paramsingh** [comment]
Hi @kartikeyaSh, we'd be really grateful if you could rebase this once please?

**Comment by paramsingh** [comment]
Closing this as the repository has been merged into [listenbrainz-server](https://github.com/metabrainz/listenbrainz-server).
